src/extensionsIntegrated/CSSColorPreview/main.js

```diff
@@ -53,6 +53,25 @@ define(function (require, exports, module) {
         description: Strings.DESCRIPTION_CSS_COLOR_PREVIEW
     });
 
+
+    /**
+     * Helper function to find the nth occurrence of a substring in a string
+     * @param {String} str the line text
+     * @param {String} substr color
+     * @param {Number} nth the occurence count of the color
+     * @return {Number} the index of the nth occurence
+     */
+    function findNthOccurrence(str, substr, nth) {
+        let pos = -1;
+        while (nth-- > 0) {
+            pos = str.indexOf(substr, pos + 1);
```

Comment:
individual string character extraction and matching is very costly, so better to avoid when possible. Prefer regex or indexOf("someString") or direct high lever APIs when possible.

In this particular use case, this can be avoided. See below comment.